OneSignalSDK/onesignal/notifications/src/main/java/com/onesignal/NotificationOpenedActivityHMS.kt

```diff
@@ -72,13 +72,14 @@ class NotificationOpenedActivityHMS : Activity() {
     }
 
     private fun processOpen(intent: Intent?) {
-        if (!OneSignal.initWithContext(applicationContext)) {
-            return
-        }
+        // run init in background
+        suspendifyOnThread {
```

Comment:
I have concerns this is going to have side-effects. Mainly that `notificationPayloadProcessorHMS.handleHMSNotificationOpenIntent` will now run AFTER `finish()`. I recommend use the same logic from `NotificationOpenedActivityBase` to avoid changing order of execution.